### ISSUE TITLE
Common Intent called by different steps?

### ISSUE THREAD

**Issue by gadams999** [dgtony/afg#1]
Hi @dgtony,

Great work on this addition to `flask-ask`. I'm trying to craft a conversation with various prompts, and have a question around single work utterances. Near the start of my state table I ask for a specific month, then later in the state I ask for a specific date (M/D/Y). Both requests are expecting just a single value, which I assume I would map to `AMAZON.DATE`. Is there a way to map a common intent and have it redirect to the next step based on the who called (previous step)?

Something like this where `DateChoiceIntent` is the common one:

`[init] --> [date_choice] --> <choose_month> --> [foo_choice] --> ... --> [date_choice] -->
 <choose_full_date. --> [bar_choice]`

Basically where if only a month is returned to DateChoiceIntent, date_choice() would forward to foo_choice() state. If a full date is returned, date_choice() would forward to the bar_choice() state.

**Comment by dgtony** [comment]
Hi, @gadams999!
You can separate intent handling from scenario step verification and such verification could be divided in two cases: month and full date. Scenario may look like this:

```yaml
first_step: init
last_step: done
default_help: "Test skill"

steps:
  init:
    events:
      launched:
        next: date_choice

  date_choice:
    reprompt: "Repeat the date please"
    events:
      choose_month:
        next: foo_choice
      choose_full_date:
        next: bar_choice

  foo_choice:
    reprompt: "Did you mean foo?"
    events:
      choose_foo:
        next: done

  bar_choice:
    reprompt: "Did you mean bar?"
    events:
      choose_bar:
        next: done

  done:
```

I've tried it on a following simple skill.

```python
from flask import Flask
from flask_ask import Ask, statement, question
from afg import Supervisor
import logging


app = Flask("Alexa bot")
ask = Ask(app, '/')
sup = Supervisor("scenario.yaml")


@ask.on_session_started
@sup.start
def new_session():
    app.logger.debug('new session started')


@sup.stop
def close_user_session():
    logger.debug("user session stopped")


@ask.intent('AMAZON.CancelIntent')
def stop():
    close_user_session()
    return statement("Bye-bye")


@ask.session_ended
def session_ended():
    close_user_session()
    return "", 200


@ask.launch
@sup.guide
def launched():
    return question("Tell me some date")


@ask.intent('DateChoiceIntent')
def choose_date(date):
    # initial date classification
    date_components = date.split('-')
    if len(date_components) == 2:
        return choose_month(*date_components)
    elif len(date_components) == 3:
        return choose_full_date(*date_components)
    else:
        logger.warning("Unknown date format: {}".format(date_components))
        return question("Repeat date again, please")


@sup.guide
def choose_month(year, month):
    logger.info("Choose month: {} (year: {})".format(month, year))
    return question("Your input is month, now try to choose foo or bar")


@sup.guide
def choose_full_date(year, month, day):
    logger.info("Choose full date. Year: {}, month: {}, day: {}".format(year, month, day))
    return question("Your input is full date, now try to choose foo or bar")


@ask.intent('FooIntent')
@sup.guide
def choose_foo():
    return statement("Your choice is foo")


@ask.intent('BarIntent')
@sup.guide
def choose_bar():
    return statement("Your choice is bar")


if __name__ == "__main__":
    logging.basicConfig(level=logging.DEBUG)
    logger = logging.getLogger('Skill')
    app.run(host='127.0.0.1', port=1234, debug=True)
```

As you can see intent handler function for `DateChoiceIntent` (data type: `AMAZON.DATE`) is `choose_date`, while step verification was placed in functions `choose_month` and `choose_full_date` depending on the results of user input validation.


BTW, thanks for the interesting question, if you don't mind I'll add this case in readme.

**Comment by gadams999** [comment]
Ah, I think I understand. Basically you create an Intent that is not managed by the conversation/state flow. Based on the response (YYYY-MM or YYYY-MM-DD) you then return those components to the intent that is under `afg` supervisor control? Appreciate the sample code, will keep it in mind.

Please feel free to add this to the README!
